### PR TITLE
Hw6 cucumber

### PR DESCRIPTION
**Pull request by vadikpkin**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -120,6 +132,27 @@
...
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-testng</artifactId>
+            <version>${cucumber.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-core</artifactId>
+            <version>5.6.0</version>
```
Comment (DmitryKhodakovsky) [inline pom.xml]: You have property for the cucumber version.

File: src/test/java/hw6/steps/ThenSteps.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class ThenSteps {
+
+    private DifferentElementsPage differentElementsPage;
+    private UserTablePage userTablePage;
+
+    @Then("For each checkboxes {string} and {string} there is an individual log row and value is corresponded to the status of checkbox")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: How will you manage if you will need add one more checkbox?

File: src/test/java/hw6/steps/ThenSteps.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    @Then("{int} checkboxes should be displayed on Users Table on User Table Page")
+    public void iAssertCheckboxesOnUserTablePage(int expectedCheckboxesAmount) {
+        assertEquals(userTablePage.getUsersCheckboxes().size(), 6);
+        assertTrue(userTablePage.isUsersCheckboxesDisplayed());
+    }
+
+    @Then("User table should contain following values:")
+    public void iAssertContentOfTableOnUserTablePAge(DataTable dataTable) {
+        List<UserTableRow> actualUserTable = CalculationUtils
+                .convertToUserTableRowList(userTablePage.getUsersIds(), userTablePage.getUsersNames(), userTablePage.getUsersImagesDescriptions());
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: Why `convertToUserTableRowList` is in the `CalculationUtils`? Also Cucumber has mechanism which allows set `List<UserTableRow>` as parameter to the step

File: src/test/java/hw6/steps/WhenSteps.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    private DifferentElementsPage differentElementsPage;
+    private UserTablePage userTablePage;
+
+    @When("I select checkboxes {string}, {string}")
+    public void iSelectCheckboxesOnDifferentElementsPage(String firstCheckbox, String secondCheckbox) {
+        differentElementsPage = new DifferentElementsPage(WebDriverSingleton.INSTANCE.getDriver());
+        differentElementsPage.setCheckBox(firstCheckbox);
+        differentElementsPage.setCheckBox(secondCheckbox);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/WhenSteps.java]: How will you manage if you will need check only one checkbox?

File: src/main/java/utils/CalculationUtils.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    public static List<UserTableRow> convertUserDataToList(DataTable dataTable) {
+        List<UserTableRow> expectedUserTable = new ArrayList<>();
+        List<List<String>> dataTableRows = dataTable.asLists(String.class);
+        for (List<String> columns : dataTableRows) {
+            expectedUserTable.add(new UserTableRow(columns.get(0), columns.get(1), columns.get(2)));
+        }
+        //to remove columns Titles
+        expectedUserTable.remove(0);
+        return expectedUserTable;
+    }
```
Comment (DmitryKhodakovsky) [inline src/main/java/utils/CalculationUtils.java]: I am not sure that this is correct class for the current methods

File: src/main/java/utils/CalculationUtils.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    public static List<UserTableRow> convertToUserTableRowList(List<String> numbers, List<String> usernames, List<String> description) {
+        List<UserTableRow> userTable = new ArrayList<>();
+        for (int i = 0; i < numbers.size(); i++) {
+            userTable.add(new UserTableRow(numbers.get(i), usernames.get(i), description.get(i)));
+        }
+        return userTable;
+    }
+
+    public static List<UserTableRow> convertUserDataToList(DataTable dataTable) {
```
Comment (DmitryKhodakovsky) [inline src/main/java/utils/CalculationUtils.java]: Why do you pass `DataTable` as parameter here?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Resolve comments

File: src/test/java/hw6/steps/ThenSteps.java (hunk 2 above)
Comment (vadikpkin) [reply src/test/java/hw6/steps/ThenSteps.java]: added list

File: src/test/java/hw6/steps/WhenSteps.java (hunk 4 above)
Comment (vadikpkin) [reply src/test/java/hw6/steps/WhenSteps.java]: added list

File: src/main/java/utils/CalculationUtils.java (hunk 5 above)
Comment (vadikpkin) [reply src/main/java/utils/CalculationUtils.java]: thanks for advice, replaced to test

File: src/main/java/utils/CalculationUtils.java (hunk 6 above)
Comment (vadikpkin) [reply src/main/java/utils/CalculationUtils.java]: remade all that stuff with the help of POJO from params

File: src/test/java/hw6/steps/ThenSteps.java (hunk 3 above)
Comment (vadikpkin) [reply src/test/java/hw6/steps/ThenSteps.java]: remade all that stuff with the help of POJO from params

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved

**Pull request merged**